### PR TITLE
Adding rust/js templates for local pickup generators

### PR DESCRIPTION
**Pull request by stevenpslade**
Issue: https://github.com/Shopify/shopify/issues/446579

This PR adds the Javascript and Rust function templates for the local pickup delivery option generator functions API. This work is needed before adding the API type to partners here: https://github.com/Shopify/partners/pull/49628

### CODE REVIEW

**Comment by stevenpslade** [comment]
The following error is occurring in the rust checks of this PR:
```
error[E0412]: cannot find type `Handle` in module `super`
 --> order-routing/rust/local-pickup-delivery-option-generators/default/src/main.rs:6:1
  |
6 | / generate_types!(
7 | |     query_path = "./input.graphql",
8 | |     schema_path = "./schema.graphql"
9 | | );
  | |_^ not found in `super`
```

However, my assumption is that `Handle` should be recognized after this PR was merged: https://github.com/Shopify/shopify-function-rust/pull/46/files

Maybe the version of `shopify-function-rust` is different in the CI running the test?

File: order-routing/rust/local-pickup-delivery-option-generators/default/input.graphql.liquid [hunk 1]
```diff
@@ -0,0 +1,29 @@
+query Input {
```
Comment (joshua-arts) [inline order-routing/rust/local-pickup-delivery-option-generators/default/input.graphql.liquid:1]: Does this need to be a `.liquid` file if there is no liquid syntax?

File: order-routing/rust/local-pickup-delivery-option-generators/default/src/main.rs [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+);
+
+#[derive(Serialize, Deserialize, Default, PartialEq)]
+struct Config {}
+
+#[shopify_function]
+fn function(_input: input::ResponseData) -> Result<output::FunctionResult> {
+    let operations = vec![];
+
+    // Build operations based on the input query response here.
```
Comment (nargesashrafizadeh) [inline order-routing/rust/local-pickup-delivery-option-generators/default/src/main.rs:18]: Can this actually return a sample "add" operation?

Comment (stevenpslade) [reply order-routing/rust/local-pickup-delivery-option-generators/default/src/main.rs:30]: Done ✅

**Comment by nickwesselman** [comment]
Please be sure to add a wasm template as well, if not now then before public release of this API.

**Pull request merged**